internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package extendedlocation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/extendedlocation/2021-08-15/customlocations"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CustomLocationResource struct{}
+
+type CustomLocationResourceModel struct {
+	Name                string      `tfschema:"name"`
+	ResourceGroupName   string      `tfschema:"resource_group_name"`
+	Location            string      `tfschema:"location"`
+	Authentication      []AuthModel `tfschema:"authentication"`
+	ClusterExtensionIds []string    `tfschema:"cluster_extension_ids"`
+	DisplayName         string      `tfschema:"display_name"`
+	HostResourceId      string      `tfschema:"host_resource_id"`
+	HostType            string      `tfschema:"host_type"`
+	Namespace           string      `tfschema:"namespace"`
+}
+
+type AuthModel struct {
+	Type  string `tfschema:"type"`
+	Value string `tfschema:"value"`
+}
+
+func (r CustomLocationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Can we improve this validation? There must be some naming rules here?

Reply:
passing through, I remembered the portal has this:
<img width="807" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/11830746/d7ee67e4-7d9b-4cdd-a3e8-c7c65da2ba47">

Reply:
Updated.